lib/rbac/filterer.rb

```diff
@@ -254,10 +259,26 @@ def search(options = {})
 
       scope = include_references(scope, klass, include_for_find, exp_includes, skip_references)
       scope = scope.limit(limit).offset(offset) if attrs[:apply_limit_in_sql]
+
+      # putting this into an inner view will allow us to apply the virtual attributes to LIMIT rows not all of them.
+      if options[:use_sql_view]
+        if attrs[:apply_limit_in_sql] &&
+           limit && extra_cols &&
+           !klass.table_name&.include?(".") && scope.respond_to?(:includes_values)
+          inner_scope = scope.except(:select, :includes, :references)
+          scope.includes_values.each { |hash| inner_scope = add_joins(klass, inner_scope, hash) }
```

Comment:
Yea, the “I know you said to in-line this but it will bread under these conditions” block may be too cautious.

The if block reads: if this is enabled, and it would benifit us, and it would not blowup, then...

And the inline view reads: convert this to just select from the primary table without the extra tables (but do left join to tables so we can use the where clause)

And the main scope no longer needs the where and limit, since those are already applied in the inline view.

I thought this was getting pretty minimal. I agree that the conversion from requires to joins is not optimal, but that is the only complexity there.

Taking a step back, is the applying of the join your main gripe?

Reply:
In the perfect scenario, the inline view is simple:

```sql
FROM (select * from services limit 20) as services
```

When the references stay there, it throws wrenches into the mix. So many different ways. Converting to joins ended up being simpler than handling all the edge cases.

That was the reason I had the select(“*”) in there before. But it broke too easily with different eBay filters.

So while not optimal, this was the simplest way I could get it consistently working without thinking a different type of rbac filter would break things. And it gave me the most confidence.

Reply:
> Taking a step back, is the applying of the join your main gripe?

Yes, but only in the context of a backport.  Goes back to what I was saying in the "review description", but from the info I have provided from you, I can't see where this is needed to fix the BZ.

I definitely can see why this is needed a broader fix outside the scope of `Service.yaml`, but I haven't been presented with any case within that scope where doing the `add_joins` is necessary to specifically fix the BZ.  The only reason I have been able determine you added the `add_joins` was to fix specs and use cases when `options[:use_sql_view]` was always enabled, but again, I think that falls outside the scope of a "targeted bugfix".

So I would like to drop it for this PR (if possible), but my gripe is only that it "adds risk/complexity" that isn't needed **now** (please correct me if I am wrong on this), not that "it is bad, never do this".

That said, we **should** add this as part of the follow up PR we talked about if it turns out we don't require it here, and I will have next to no problem with that there.  And in regards to my suggestion previously:

> I am also wondering if it makes sense to undo what caused this issue in the first place, which was a PR that I wrote three years ago: https://github.com/ManageIQ/manageiq/pull/11502

Is probably much more work and haven't fully vetted, just something I thought as a _potential_ alternative.

Reply:
> When the references stay there, it throws wrenches into the mix. So many different ways. Converting to joins ended up being simpler than handling all the edge cases.
> 
> That was the reason I had the select(“*”) in there before. But it broke too easily with different rbac filters.

So these were the edge cases I was curious about when I said:

> ... and if there are any use cases that aren't described in the PR that some of the lines of code are addressing (`add_joins` for example), then we should document those in this PR and/or in specs.

If it broke when using filters on `Service.yaml`, you basically can ignore any gripes I had with this.  However, I would like some examples (specs, documentation, and/or PR description update), since I am still unclear if those Rbac filters are use cases relevant to `Service.yaml`.

Reply:
I think it is necessary for `Service.yaml` BUT I need to create a spec to prove it.
I don't know all the parameters necessary to really tweak out rbac. It is possible that the gook I'm trying to avoid is not present for `Service`

Thanks for keeping me honest here. I really appreciate it.

Reply:
Yea, it wasn't that hard to reproduce the problem here. Turns out a simple miq_expression forced the error.